lib/clib/Corres_UL_C.thy

```diff
@@ -56,6 +56,20 @@ lemma exec_handlers_use_hoare_nothrow:
   apply simp
   done
 
+lemma exec_handlers_use_hoare_nothrow_hoarep:
+  assumes ce: "E \<turnstile>\<^sub>h \<langle>c # hs, s'\<rangle> \<Rightarrow> (n, t)"
+  and   asms: "s' \<in> R'"
+  and valid': "E \<turnstile> R' c Q'"
+  shows "E \<turnstile> \<langle>c, Normal s'\<rangle> \<Rightarrow> t \<and> isNormal t"
+  apply (insert assms)
+  apply (drule hoare_sound)
+  apply (clarsimp simp: cvalid_def HoarePartialDef.valid_def)
+  apply (erule exec_handlers.cases)
```

Comment:
fastforce should be able to handle everything from here, possibly the erule as well when given as `elim!`